plugins/CoreAdminHome/CustomLogo.php

```diff
@@ -163,6 +163,16 @@ protected static function rewritePath($path)
         return SettingsPiwik::rewriteMiscUserPathWithInstanceId($path);
     }
 
+    public static function hasUserLogo()
+    {
+        return file_exists(Filesystem::getPathToPiwikRoot() . '/' . CustomLogo::getPathUserLogo());
```

Comment:
Somehow I would prefer not to have it in Filesystem but would be also ok there. Having a look at this again in the code there is often `file_exists($pathToPiwikRoot . '/' . $customLogo)` etc, finding several `file_exists`. The code seems quite confusing in there in general and needs to be refactored at some point. Maybe there could me instead a method
```php
private function logoExists($reference)
{
    return file_exists(Filesystem::getPathToPiwikRoot() . '/' . $reference);
}
```
Of course this could be as well in Filesystem but might be as well fine in CustomLogo for now